src/app/campaign/form/campaign-form.component.ts

```diff
@@ -87,8 +87,11 @@ export class CampaignFormComponent implements OnInit {
 
   updateCampaignForm(campaign: Campaign) {
     const { name, type, repName, notes, set_account_uri, set_advertiser_uri } = campaign;
-    const findAdvertiserByURI =
-      set_advertiser_uri && this.advertisers && this.advertisers.find(adv => adv.set_advertiser_uri === set_advertiser_uri);
+
+    // set_advertiser_uri might be a typeahead advertiser name instead
+    // NOTE: this fails if someone types a url as the advertiser name
+    const isAdvertiserName = set_advertiser_uri && !set_advertiser_uri.match(/^http(s)?:.+\/api\/v1\//);
```

Comment:
Makes sense. I wonder if we can proxy this field in the future somehow?  Like a `advertiser_name` field that proxies to the `set_advertiser_uri` key. Looks good!

Reply:
+1.  I didn't dig too far into that autocomplete component ... but it would be more straightforward if it didn't reuse this field to store whatever partial string you've typed.